### PR TITLE
feat: make `srcset` actually useful

### LINKED ISSUE #1168: [FEATURE] Don't cache remote image link when cacheRemoteImages is enabled
### Describe the feature you want 描述你的功能需求

The image link in markdown and cacheRemoteImages is enabled,  The image for the link to be cached.
for examples:
```
[xxx](https://en.wikipedia.org/wiki/Netfilter#/media/File:Netfilter-packet-flow.svg)
```
Beacuse not only https://github.com/HEIGE-PCloud/DoIt/blob/main/layouts/_default/_markup/render-image.html  call  "function/resource.html" but also https://github.com/HEIGE-PCloud/DoIt/blob/main/layouts/_default/_markup/render-link.html#L7  also call "function/resource.html"
https://github.com/HEIGE-PCloud/DoIt/blob/main/layouts/partials/function/getRemoteImage.html

### Useful reference 有价值的参考

_No response_

### PR DESCRIPTION
**Pull request by 713b76b7-2247-4bd5-98a7-014226363f9b**
also fix #1168 

will work on documentation after everything is settled.

### CODE REVIEW

File: layouts/partials/function/resource.html
Comment (713b76b7-2247-4bd5-98a7-014226363f9b) [inline layouts/partials/function/resource.html:1]: Just revert the changes I made in `cacheRemoteImages` pr, many other partial will use this hook, would be better to keep it "neutral".

File: layouts/partials/plugin/image.html
Comment (713b76b7-2247-4bd5-98a7-014226363f9b) [inline layouts/partials/plugin/image.html:1]: To avoid interfering with the images in the header and other sections, I have created two variables: ﻿`Optim` and ﻿`Remote`. To enable the corresponding functionality, you must explicitly set these variables to true. 

The srcset optimization is now enabled in the following parts: 
* markdown image render hook
* image shortcode
* summary image in single view
* summary image in list view.

File: exampleSite/config/_default/params.toml [hunk 1]
```diff
@@ -27,9 +27,17 @@ enablePWA = false
...
-cacheRemoteImages = false
+cacheRemoteImages = true
+# [Experimental] generate image srcset attribute with hugo image processing feature.
+# [试验性功能] 使用 hugo 图片处理功能生成 srcset 属性
+optimizeImages = true
+# [Experimental] resize method for srcset attribute in optimizeImages, more info: https://gohugo.io/content-management/image-processing/#image-processing-options
+# [实验性功能] optimizeImages 中 srcset 属性的缩放方法, 详情请见：https://gohugo.io/content-management/image-processing/#image-processing-options
+srcsetSmallResizeMethod = "700x webp Lanczos q75"
+srcsetDefaultResizeMethod = "1200x webp Lanczos q75"
+srcsetLargeResizeMethod = "2000x webp Lanczos q75"
```
Comment (HEIGE-PCloud) [inline exampleSite/config/_default/params.toml]: Will it be better if we moves these options under `[page]` so there can be page level control using the front matter?

**Review by HEIGE-PCloud — APPROVED** [review]
